### ISSUE TITLE
label encoder

### ISSUE THREAD

**Issue by suruchi18coder** [suruchi18coder/machine_learning#1]
I have started a machine learning course on udemy. I was writing the preprocessing code.
my code :
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

dataset= pd.read_csv('samp_sheet.csv')
x=dataset.iloc[:,:-1]
z= dataset.iloc[:,3]

from sklearn.preprocessing import OneHotEncoder, LabelEncoder
labelencoder= LabelEncoder()
x.values[:,0]=labelencoder.fit_transform(x.values[:,0])

I didn't get any changes in my dataset 
![screen shot 2018-11-21 at 1 45 11 pm](https://user-images.githubusercontent.com/41668712/48827614-b48e4c00-ed93-11e8-9c7a-066ca1a9081d.png)

but when I used 
x=dataset.iloc[:,:-1].values
z= dataset.iloc[:,3].values

from sklearn.preprocessing import OneHotEncoder, LabelEncoder
labelencoder= LabelEncoder()
x[:,0]=labelencoder.fit_transform(x[:,0])

the output was:
array([1, 1, 2, 1, 0, 2, 0, 1, 2, 0], dtype=object)

why?
